main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ShapeIncreasingDistanceValidator.java

```diff
@@ -47,57 +56,119 @@ public void validate(NoticeContainer noticeContainer) {
       for (int i = 1; i < shapeList.size(); ++i) {
         GtfsShape prev = shapeList.get(i - 1);
         GtfsShape curr = shapeList.get(i);
-        if (prev.hasShapeDistTraveled()
-            && curr.hasShapeDistTraveled()
-            && prev.shapeDistTraveled() >= curr.shapeDistTraveled()) {
+        if (!(prev.hasShapeDistTraveled() || curr.hasShapeDistTraveled())) {
```

Comment:
Not quite sure, but do you want to verify that both `shape_dist_traveled` are present before validating them? If yes, the condition should be `!(prev.hasShapeDistTraveled() && curr.hasShapeDistTraveled())` because you want it to be false only if both `shape_dist_traveled` are present, not one or the other. In the current state, if `prev.hasShapeDistTraveled() == true` and `curr.hasShapeDistTraveled() == false`, you would have `(prev.hasShapeDistTraveled() || curr.hasShapeDistTraveled()) == (true || false) == (true)` and `!(true) == false`. We would end up validating empty/null `shape_dist_traveled` values, so potential problems here.

Reply:
Good catch! Thanks.